provider/infoblox.go

```diff
@@ -0,0 +1,483 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	ibclient "github.com/khrisrichardson/infoblox-go-client"
+	"github.com/kubernetes-incubator/external-dns/endpoint"
+	"github.com/kubernetes-incubator/external-dns/plan"
+)
+
+// InfobloxConfig clarifies the method signature
+type InfobloxConfig struct {
+	DomainFilter DomainFilter
+	Host         string
+	Port         int
+	Username     string
+	Password     string
+	Version      string
+	SSLVerify    bool
+	DryRun       bool
+}
+
+// InfobloxProvider implements the DNS provider for Infoblox.
+type InfobloxProvider struct {
+	client       ibclient.IBConnector
+	domainFilter DomainFilter
+	dryRun       bool
+}
+
+type ibObjRes struct {
```

Comment:
ibObjRes?
It would be nice to have a better name here.
And the best would be to refactor this code to have some methods on top of this new struct here, such that it simplifies your create/update/delete provider methods. Please see some other comments that will follow.

Reply:
@szuecs I'm not exactly sure why, but I get the impression you're not crazy about the name ⚡️